scipy/integrate/_cubature.py

```diff
@@ -374,3 +469,226 @@ def _process_subregion(data):
+
+    If :math:`a_i` and :math:`b_i` range over :math:`x \in (-\infty, \infty)`, the i-th
+    integration variable will use the transformation :math:`x = \frac{1-|t|}{t}` and
+    :math:`t \in (-1, 1)`.
+
+    If :math:`a_i` and :math:`b_i` range over :math:`x \in [a_i, \infty)`, the i-th
+    integration variable will use the transformation :math:`x = a_i + \frac{1-t}{t}` and
+    :math:`t \in (0, 1)`.
+
+    If :math:`a_i` and :math:`b_i` range over :math:`x \in (-\infty, b_i]`, the i-th
+    integration variable will use the transformation :math:`x = b_i - \frac{1+t}{t}` and
+    :math:`t \in (0, 1)`.
+
+    In all three of these cases, the Jacobian of the transformation is
+    :math:`J(t) = t^{-2}`.
+    """
+
+    def __init__(self, f, a, b):
+        self._xp = array_namespace(a, b)
+
+        self._f = f
+        self._orig_a = a
+        self._orig_b = b
+
+        self._negate_pos = []
+        self._semi_inf_pos = []
+        self._double_inf_pos = []
+
+        ndim = xp_size(a)
+
+        for i in range(ndim):
+            if a[i] == -math.inf and b[i] == math.inf:
+                # (-oo, oo) will be mapped to (-1, 1).
+                self._double_inf_pos.append(i)
+
+            elif a[i] != -math.inf and b[i] == math.inf:
+                # (start, oo) will be mapped to (0, 1).
+                self._semi_inf_pos.append(i)
+
+            elif a[i] == -math.inf and b[i] != math.inf:
+                # (-oo, end) will be mapped to (0, 1).
+                #
+                # This is handled by making the transformation t = -x and reducing it to
+                # the other semi-infinite case.
+                self._negate_pos.append(i)
+                self._semi_inf_pos.append(i)
+
+                # Since we flip the limits, we don't need to separately multiply the
+                # integrand by -1.
+                self._orig_a[i] = -b[i]
+                self._orig_b[i] = -a[i]
+
+    @property
+    def transformed_limits(self):
+        a = xp_copy(self._orig_a)
+        b = xp_copy(self._orig_b)
+
+        for index in self._double_inf_pos:
+            a[index] = -1
+            b[index] = 1
+
+        for index in self._semi_inf_pos:
+            a[index] = 0
+            b[index] = 1
+
+        return a, b
+
+    @property
+    def points(self):
+        # If there are infinite limits, then the origin becomes a problematic point
+        # due to a division by zero there.
+        if len(self._double_inf_pos) != 0 or len(self._semi_inf_pos) != 0:
+            return [self._xp.zeros(self._orig_a.shape)]
+        else:
+            return []
+
+    def inv(self, x):
+        t = xp_copy(x)
+
+        for i in self._negate_pos:
+            t[..., i] *= -1
+
+        for i in self._double_inf_pos:
+            t[..., i] = 1/(x[..., i] + self._xp.sign(x[..., i]))
+
+        for i in self._semi_inf_pos:
+            t[..., i] = 1/(x[..., i] - self._orig_a[i] + 1)
```

Comment:
Again these are arrays right so we can skip the loop and use a list of indexes

Reply:
I've tried this but am getting errors using `array_api_strict`. Switching to boolean indexing also doesn't work since `t[..., self._semi_inf_pos]` raises errors about not being able to use boolean indexing and `...` at the same time.